<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>InteRD • InteRD</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="InteRD">
<meta property="og:description" content="InteRD">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">InteRD</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li>
  <a href="../articles/NAME-OF-VIGNETTE.html">Tutorial</a>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li>
  <a href="../articles/web/data.html">Data</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/chencxxy28/InteRD/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>InteRD</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/chencxxy28/InteRD/blob/HEAD/vignettes/NAME-OF-VIGNETTE.Rmd" class="external-link"><code>vignettes/NAME-OF-VIGNETTE.Rmd</code></a></small>
      <div class="hidden name"><code>NAME-OF-VIGNETTE.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://devtools.r-lib.org/" class="external-link">"devtools"</a></span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"chencxxy28/InteRD"</span><span class="op">)</span></span></code></pre></div>
<p>In this tutorial, we use SCDC (<a href="https://academic.oup.com/bib/article/22/1/416/5699815" class="external-link">Dong et
al. (2021)</a>) for illustration to conduct reference-based
deconvolution.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#load</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/chencxxy28/InteRD" class="external-link">InteRD</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="implementation-case-study-1">Implementation (Case study 1)<a class="anchor" aria-label="anchor" href="#implementation-case-study-1"></a>
</h2>
<p>First, read in scRNA-seq data from our website</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">readRDSFromWeb</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">ref</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/gzcon.html" class="external-link">gzcon</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="va">ref</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">seger</span> <span class="op">&lt;-</span> <span class="fu">readRDSFromWeb</span><span class="op">(</span><span class="st">"https://github.com/chencxxy28/Data/raw/main/data_InteRD/segerstolpe.rds"</span><span class="op">)</span></span>
<span><span class="va">baron</span> <span class="op">&lt;-</span> <span class="fu">readRDSFromWeb</span><span class="op">(</span><span class="st">"https://github.com/chencxxy28/Data/raw/main/data_InteRD/baron.rds"</span><span class="op">)</span></span>
<span><span class="va">xin</span><span class="op">&lt;-</span><span class="fu">readRDSFromWeb</span><span class="op">(</span><span class="st">"https://github.com/chencxxy28/Data/raw/main/data_InteRD/Xin_nonD.rds"</span><span class="op">)</span></span></code></pre></div>
<p>Second, use function <code><a href="../reference/generateBulk.html">generateBulk()</a></code> to create the
<em>pseudo bulk</em> samples, where <code>ct.varname</code> specifies
the name of cell type clustering result variable. <code>sample</code>
specifies the name of subjects information variable. <code>ct.sub</code>
specifies the names of cell types used to construct <em>pseudo bulk</em>
samples. Here we provide an example where the <em>pseudo bulk</em>
samples are generated from <a href="https://www.sciencedirect.com/science/article/pii/S1550413116304363?via%3Dihub" class="external-link">Segerstolpe
et al. (2016)</a>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1234567</span><span class="op">)</span></span>
<span><span class="va">pseudo.seger</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/generateBulk.html">generateBulk</a></span><span class="op">(</span><span class="va">seger</span><span class="op">[[</span><span class="st">"sc.eset.qc"</span><span class="op">]</span><span class="op">]</span>, ct.varname <span class="op">=</span> <span class="st">"cluster"</span>, sample <span class="op">=</span> <span class="st">"sample"</span>, ct.sub <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"alpha"</span>,<span class="st">"beta"</span>,<span class="st">"delta"</span>,<span class="st">"gamma"</span><span class="op">)</span>, nbulk <span class="op">=</span> <span class="fl">40</span>, low_s <span class="op">=</span> <span class="fl">0.3</span>, upp_s <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span></span>
<span><span class="va">truep</span><span class="op">&lt;-</span><span class="va">pseudo.seger</span><span class="op">$</span><span class="va">true_p</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/stats/complete.cases.html" class="external-link">complete.cases</a></span><span class="op">(</span><span class="va">pseudo.seger</span><span class="op">$</span><span class="va">true_p</span><span class="op">)</span>,<span class="op">]</span></span></code></pre></div>
<p>The generated <em>pseudo bulk</em> object contains a matrix of true
cell type proportions (<code>pseudo.seger$truep</code>) and the
<code>ExpressionSet</code> object
(<code>pseudo.seger$pseudo_eset</code>).</p>
<p>InteRD1 algorithm is the first step of InteRD to do multiple
reference ensemble, which takes as input the target bulk data, a list of
marker genes (pre-selected by <code>Seurat</code> based on a single cell
data from <a href="https://www.sciencedirect.com/science/article/pii/S155041311630434X?via%3Dihub" class="external-link">Xin
et al. (2016)</a>), a list of subject-level cell type proportions based
on each reference set. For illustration, we use the function
<code><a href="https://rdrr.io/pkg/SCDC/man/SCDC_ENSEMBLE.html" class="external-link">SCDC_ENSEMBLE()</a></code> from SCDC package (<a href="https://academic.oup.com/bib/article/22/1/416/5699815" class="external-link">Dong et
al. (2021)</a>) to obtain single-reference-based estimates and ensemble
estimates. For reference panels, we consider two single cell data, one
from <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5228327/" class="external-link">Baron et
al. (2016)</a>, and the other from <a href="https://www.sciencedirect.com/science/article/pii/S155041311630434X?via%3Dihub" class="external-link">Xin
et al. (2016)</a>. The estimates can be obtained by running the
following code</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1234567</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/meichendong/SCDC" class="external-link">SCDC</a></span><span class="op">)</span></span>
<span><span class="co">##ensemble of multiple reference sets </span></span>
<span><span class="co">#resuts based on SCDC</span></span>
<span>  <span class="va">pancreas.sc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>baron <span class="op">=</span> <span class="va">baron</span><span class="op">$</span><span class="va">sc.eset.qc</span>,</span>
<span>                      xin   <span class="op">=</span> <span class="va">xin</span></span>
<span>  <span class="op">)</span></span>
<span>  <span class="va">SCDC_results</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/SCDC/man/SCDC_ENSEMBLE.html" class="external-link">SCDC_ENSEMBLE</a></span><span class="op">(</span>bulk.eset <span class="op">=</span> <span class="va">pseudo.seger</span><span class="op">$</span><span class="va">pseudo_eset</span>, sc.eset.list <span class="op">=</span> <span class="va">pancreas.sc</span>, ct.varname <span class="op">=</span> <span class="st">"cluster"</span>,</span>
<span>                                sample <span class="op">=</span> <span class="st">"sample"</span>, weight.basis <span class="op">=</span><span class="cn">TRUE</span>,truep <span class="op">=</span> <span class="va">truep</span>, ct.sub <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"alpha"</span>,<span class="st">"beta"</span>,<span class="st">"delta"</span>,<span class="st">"gamma"</span><span class="op">)</span>, search.length <span class="op">=</span> <span class="fl">0.02</span>,grid.search<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span>  <span class="va">comb</span><span class="op">&lt;-</span><span class="va">SCDC_results</span><span class="op">$</span><span class="va">prop.only</span></span>
<span>  <span class="va">weight_matrix</span><span class="op">&lt;-</span><span class="va">SCDC_results</span><span class="op">$</span><span class="va">w_table</span><span class="op">[</span><span class="st">"mAD_Y"</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span>
<span>  <span class="va">SCDC_ENSEMBLE_MAD</span><span class="op">&lt;-</span><span class="fu">SCDC</span><span class="fu">:::</span><span class="fu"><a href="https://rdrr.io/pkg/SCDC/man/wt_prop.html" class="external-link">wt_prop</a></span><span class="op">(</span><span class="va">weight_matrix</span>,<span class="va">comb</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># saveRDS(SCDC_ENSEMBLE_MAD,"/Users/chixiang.chen/Library/CloudStorage/OneDrive-UniversityofMarylandSchoolofMedicine/postdoc/postdoc/deconvolution/ref_based_rd/data_InteRD/SCDC_ENSEMBLE_MAD_seger.rds")</span></span>
<span>  <span class="co"># saveRDS(comb,"/Users/chixiang.chen/Library/CloudStorage/OneDrive-UniversityofMarylandSchoolofMedicine/postdoc/postdoc/deconvolution/ref_based_rd/data_InteRD/comb_seger.rds")</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#results based on InteRD1</span></span>
<span><span class="va">SCDC_ENSEMBLE_MAD</span><span class="op">&lt;-</span><span class="fu">readRDSFromWeb</span><span class="op">(</span><span class="st">"https://github.com/chencxxy28/Data/raw/main/data_InteRD/SCDC_ENSEMBLE_MAD_seger.rds"</span><span class="op">)</span></span>
<span><span class="va">comb</span><span class="op">&lt;-</span><span class="fu">readRDSFromWeb</span><span class="op">(</span><span class="st">"https://github.com/chencxxy28/Data/raw/main/data_InteRD/comb_seger.rds"</span><span class="op">)</span></span>
<span>  <span class="va">list_marker</span><span class="op">&lt;-</span><span class="fu">readRDSFromWeb</span><span class="op">(</span><span class="st">"https://github.com/chencxxy28/Data/raw/main/data_InteRD/list_markerbaron20.rds"</span><span class="op">)</span> <span class="co">#get markers selected from xin et al (2016)</span></span>
<span>  <span class="va">lambda_option</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0.01</span>,<span class="fl">0.05</span>,<span class="fl">0.1</span>,<span class="fl">1</span>,<span class="fl">5</span>,<span class="fl">100</span><span class="op">)</span></span>
<span>  <span class="va">cell_type_unique</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"alpha"</span>,<span class="st">"beta"</span>,<span class="st">"delta"</span>,<span class="st">"gamma"</span><span class="op">)</span></span>
<span>  <span class="va">InteRD1.output</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/InteRD1.html">InteRD1</a></span><span class="op">(</span>bulk.data <span class="op">=</span><span class="va">pseudo.seger</span><span class="op">$</span><span class="va">pseudo_eset</span>,<span class="va">list_marker</span>,<span class="va">cell_type_unique</span>,comb_used<span class="op">=</span><span class="va">comb</span>,<span class="va">lambda_option</span><span class="op">)</span></span>
<span>  <span class="va">InteRD1</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/InteRD.predict.prop.html">InteRD.predict.prop</a></span><span class="op">(</span>InteRD.output<span class="op">=</span><span class="va">InteRD1.output</span><span class="op">)</span></span></code></pre></div>
<p>We provide the function <code>evaluate</code> to assess the
performance of estimated proportions versus the true proportions based
on mean absolute deviance (MAD, the smaller the better), Kendall
correlation coefficient (Ken, the larger the better), and Pearson
correlation coefficient (Cor, the larger the better).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/evaluate.html">evaluate</a></span><span class="op">(</span><span class="va">SCDC_ENSEMBLE_MAD</span>,<span class="va">pseudo.seger</span><span class="op">$</span><span class="va">true_p</span><span class="op">)</span><span class="op">$</span><span class="va">all.eva</span></span></code></pre></div>
<pre><code><span><span class="co">##     all.mad   all.ken   all.cor</span></span>
<span><span class="co">## 1 0.1069572 0.7358974 0.8939433</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/evaluate.html">evaluate</a></span><span class="op">(</span><span class="va">InteRD1</span>,<span class="va">pseudo.seger</span><span class="op">$</span><span class="va">true_p</span><span class="op">)</span><span class="op">$</span><span class="va">all.eva</span></span></code></pre></div>
<pre><code><span><span class="co">##      all.mad     all.ken    all.cor</span></span>
<span><span class="co">## 1 0.04851359 0.004487179 0.02931761</span></span></code></pre>
<p>InteRD2 algorithm is the second step of InteRD to further integrate
prior biological information into the deconvolution in a robust manner.
The prior information is the population-level mean cell-type proportions
and their corresponding standard deviations across samples. In this
tutorial, we obtain this information from scRNA-seq data from <a href="https://www.sciencedirect.com/science/article/pii/S1550413116304363?via%3Dihub" class="external-link">Segerstolpe
et al. (2016)</a>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ave_est</span> <span class="op">=</span> <span class="fu"><a href="../reference/pop.ct.prop.scRNA.html">pop.ct.prop.scRNA</a></span><span class="op">(</span>scRNA<span class="op">=</span><span class="va">seger</span><span class="op">[[</span><span class="st">"sc.eset.qc"</span><span class="op">]</span><span class="op">]</span>,cell_type_unique<span class="op">=</span><span class="va">cell_type_unique</span><span class="op">)</span><span class="op">$</span><span class="va">pop.ct.prop</span></span>
<span><span class="va">ave_sd</span> <span class="op">=</span> <span class="fu"><a href="../reference/pop.ct.prop.scRNA.html">pop.ct.prop.scRNA</a></span><span class="op">(</span>scRNA<span class="op">=</span><span class="va">seger</span><span class="op">[[</span><span class="st">"sc.eset.qc"</span><span class="op">]</span><span class="op">]</span>,cell_type_unique<span class="op">=</span><span class="va">cell_type_unique</span><span class="op">)</span><span class="op">$</span><span class="va">pop.ct.sd</span></span>
<span><span class="va">lambda_option</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span>from<span class="op">=</span><span class="fl">1</span>,to<span class="op">=</span><span class="fl">20</span>,length<span class="op">=</span><span class="fl">4</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span>from<span class="op">=</span><span class="fl">30</span>,to<span class="op">=</span><span class="fl">100</span>,length<span class="op">=</span><span class="fl">4</span><span class="op">)</span>,<span class="fl">200</span>,<span class="fl">500</span>,<span class="fl">1000000</span><span class="op">^</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">InteRD2.output</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/InteRD2.html">InteRD2</a></span><span class="op">(</span>bulk.data<span class="op">=</span><span class="va">pseudo.seger</span><span class="op">$</span><span class="va">pseudo_eset</span>,<span class="va">list_marker</span>,<span class="va">cell_type_unique</span>,comb_sampled<span class="op">=</span><span class="va">InteRD1</span>,<span class="va">ave_est</span>,<span class="va">ave_sd</span>,lambda_option<span class="op">=</span><span class="va">lambda_option</span><span class="op">)</span></span>
<span><span class="va">InteRD2</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/InteRD.predict.prop.html">InteRD.predict.prop</a></span><span class="op">(</span>InteRD.output<span class="op">=</span><span class="va">InteRD2.output</span><span class="op">)</span></span></code></pre></div>
<p>Note that Reference-free approach and TOAST are special cases of
InteRD2. The TOAST can be run based on the function <code>InteRD2</code>
by specifying <code>lambda_option=0</code>, and Reference-free approach
can be run by the function <code>Ref_free</code>. The following is the
demo for reference-free approach.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ref_free.output</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/Ref_free.html">Ref_free</a></span><span class="op">(</span>bulk.data<span class="op">=</span><span class="va">pseudo.seger</span><span class="op">$</span><span class="va">pseudo_eset</span>,list_marker<span class="op">=</span><span class="va">list_marker</span>,cell_type_unique<span class="op">=</span><span class="va">cell_type_unique</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 1</span></span></code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">reffree</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/InteRD.predict.prop.html">InteRD.predict.prop</a></span><span class="op">(</span>InteRD.output<span class="op">=</span><span class="va">ref_free.output</span><span class="op">)</span></span></code></pre></div>
<p>Based on the evaluations, we can tell that InteRD2 is better than
InteRD1, the InteRD estimates are all better than the existing
method.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/evaluate.html">evaluate</a></span><span class="op">(</span><span class="va">SCDC_ENSEMBLE_MAD</span>,<span class="va">pseudo.seger</span><span class="op">$</span><span class="va">true_p</span><span class="op">)</span><span class="op">$</span><span class="va">all.eva</span></span></code></pre></div>
<pre><code><span><span class="co">##     all.mad   all.ken   all.cor</span></span>
<span><span class="co">## 1 0.1069572 0.7358974 0.8939433</span></span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/evaluate.html">evaluate</a></span><span class="op">(</span><span class="va">reffree</span>,<span class="va">pseudo.seger</span><span class="op">$</span><span class="va">true_p</span><span class="op">)</span><span class="op">$</span><span class="va">all.eva</span></span></code></pre></div>
<pre><code><span><span class="co">##      all.mad   all.ken   all.cor</span></span>
<span><span class="co">## 1 0.03773585 0.7487179 0.9264559</span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/evaluate.html">evaluate</a></span><span class="op">(</span><span class="va">InteRD1</span>,<span class="va">pseudo.seger</span><span class="op">$</span><span class="va">true_p</span><span class="op">)</span><span class="op">$</span><span class="va">all.eva</span></span></code></pre></div>
<pre><code><span><span class="co">##      all.mad     all.ken    all.cor</span></span>
<span><span class="co">## 1 0.04851359 0.004487179 0.02931761</span></span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/evaluate.html">evaluate</a></span><span class="op">(</span><span class="va">InteRD2</span>,<span class="va">pseudo.seger</span><span class="op">$</span><span class="va">true_p</span><span class="op">)</span><span class="op">$</span><span class="va">all.eva</span></span></code></pre></div>
<pre><code><span><span class="co">##      all.mad   all.ken   all.cor</span></span>
<span><span class="co">## 1 0.01548666 0.7583333 0.9313362</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="implementation-case-study-2">Implementation (Case study 2)<a class="anchor" aria-label="anchor" href="#implementation-case-study-2"></a>
</h2>
<p>Now let us consider another pseudo data generated from <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5228327/" class="external-link">Baron et
al. (2016)</a>.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1234567</span><span class="op">)</span></span>
<span><span class="va">pseudo.baron</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/generateBulk.html">generateBulk</a></span><span class="op">(</span><span class="va">baron</span><span class="op">[[</span><span class="st">"sc.eset.qc"</span><span class="op">]</span><span class="op">]</span>, ct.varname <span class="op">=</span> <span class="st">"cluster"</span>, sample <span class="op">=</span> <span class="st">"sample"</span>, ct.sub <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"alpha"</span>,<span class="st">"beta"</span>,<span class="st">"delta"</span>,<span class="st">"gamma"</span><span class="op">)</span>, nbulk <span class="op">=</span> <span class="fl">40</span>, low_s <span class="op">=</span> <span class="fl">0.3</span>, upp_s <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span></span>
<span><span class="va">truep</span><span class="op">&lt;-</span><span class="va">pseudo.baron</span><span class="op">$</span><span class="va">true_p</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/stats/complete.cases.html" class="external-link">complete.cases</a></span><span class="op">(</span><span class="va">pseudo.baron</span><span class="op">$</span><span class="va">true_p</span><span class="op">)</span>,<span class="op">]</span></span></code></pre></div>
<p>Then we apply SCDC to obtain the estimates based on InteRD and
SCDC-ENSEMBLE, where we consider two single cell data, one from <a href="https://www.sciencedirect.com/science/article/pii/S1550413116304363?via%3Dihub" class="external-link">Segerstolpe
et al. (2016)</a>, and the other from <a href="https://www.sciencedirect.com/science/article/pii/S155041311630434X?via%3Dihub" class="external-link">Xin
et al. (2016)</a>. The estimates can be obtained by running the
following code</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1234567</span><span class="op">)</span></span>
<span><span class="co">##ensemble of multiple reference sets </span></span>
<span><span class="co">#resuts based on SCDC</span></span>
<span>  <span class="va">pancreas.sc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>seger <span class="op">=</span> <span class="va">seger</span><span class="op">$</span><span class="va">sc.eset.qc</span>,</span>
<span>                      xin   <span class="op">=</span> <span class="va">xin</span></span>
<span>  <span class="op">)</span></span>
<span>  <span class="va">SCDC_results</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/SCDC/man/SCDC_ENSEMBLE.html" class="external-link">SCDC_ENSEMBLE</a></span><span class="op">(</span>bulk.eset <span class="op">=</span> <span class="va">pseudo.baron</span><span class="op">$</span><span class="va">pseudo_eset</span>, sc.eset.list <span class="op">=</span> <span class="va">pancreas.sc</span>, ct.varname <span class="op">=</span> <span class="st">"cluster"</span>,</span>
<span>                                sample <span class="op">=</span> <span class="st">"sample"</span>, weight.basis <span class="op">=</span><span class="cn">TRUE</span>,truep <span class="op">=</span> <span class="va">truep</span>, ct.sub <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"alpha"</span>,<span class="st">"beta"</span>,<span class="st">"delta"</span>,<span class="st">"gamma"</span><span class="op">)</span>, search.length <span class="op">=</span> <span class="fl">0.02</span>,grid.search<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span>  <span class="va">comb</span><span class="op">&lt;-</span><span class="va">SCDC_results</span><span class="op">$</span><span class="va">prop.only</span></span>
<span>  <span class="va">weight_matrix</span><span class="op">&lt;-</span><span class="va">SCDC_results</span><span class="op">$</span><span class="va">w_table</span><span class="op">[</span><span class="st">"mAD_Y"</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span>
<span>  <span class="va">SCDC_ENSEMBLE_MAD</span><span class="op">&lt;-</span><span class="fu">SCDC</span><span class="fu">:::</span><span class="fu"><a href="https://rdrr.io/pkg/SCDC/man/wt_prop.html" class="external-link">wt_prop</a></span><span class="op">(</span><span class="va">weight_matrix</span>,<span class="va">comb</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># saveRDS(SCDC_ENSEMBLE_MAD,"/Users/chixiang.chen/Library/CloudStorage/OneDrive-UniversityofMarylandSchoolofMedicine/postdoc/postdoc/deconvolution/ref_based_rd/data_InteRD/SCDC_ENSEMBLE_MAD_baron.rds")</span></span>
<span>    <span class="co"># saveRDS(comb,"/Users/chixiang.chen/Library/CloudStorage/OneDrive-UniversityofMarylandSchoolofMedicine/postdoc/postdoc/deconvolution/ref_based_rd/data_InteRD/comb_baron.rds")</span></span></code></pre></div>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#results based on InteRD1</span></span>
<span><span class="va">SCDC_ENSEMBLE_MAD</span><span class="op">&lt;-</span><span class="fu">readRDSFromWeb</span><span class="op">(</span><span class="st">"https://github.com/chencxxy28/Data/raw/main/data_InteRD/SCDC_ENSEMBLE_MAD_baron.rds"</span><span class="op">)</span></span>
<span><span class="va">comb</span><span class="op">&lt;-</span><span class="fu">readRDSFromWeb</span><span class="op">(</span><span class="st">"https://github.com/chencxxy28/Data/raw/main/data_InteRD/comb_baron.rds"</span><span class="op">)</span></span>
<span>  <span class="va">list_marker</span><span class="op">&lt;-</span><span class="fu">readRDSFromWeb</span><span class="op">(</span><span class="st">"https://github.com/chencxxy28/Data/raw/main/data_InteRD/list_markerbaron20.rds"</span><span class="op">)</span> <span class="co">#get markers selected from xin et al (2016)</span></span>
<span>  <span class="va">lambda_option</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0.01</span>,<span class="fl">0.05</span>,<span class="fl">0.1</span>,<span class="fl">1</span>,<span class="fl">5</span>,<span class="fl">100</span><span class="op">)</span></span>
<span>  <span class="va">cell_type_unique</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"alpha"</span>,<span class="st">"beta"</span>,<span class="st">"delta"</span>,<span class="st">"gamma"</span><span class="op">)</span></span>
<span>  <span class="va">InteRD1.output</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/InteRD1.html">InteRD1</a></span><span class="op">(</span>bulk.data <span class="op">=</span><span class="va">pseudo.baron</span><span class="op">$</span><span class="va">pseudo_eset</span>,<span class="va">list_marker</span>,<span class="va">cell_type_unique</span>,comb_used<span class="op">=</span><span class="va">comb</span>,<span class="va">lambda_option</span><span class="op">)</span></span>
<span>  <span class="va">InteRD1</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/InteRD.predict.prop.html">InteRD.predict.prop</a></span><span class="op">(</span>InteRD.output<span class="op">=</span><span class="va">InteRD1.output</span><span class="op">)</span></span></code></pre></div>
<p>After obtaining InteRD1, we can obtain InteRD2 by integrating prior
biological information into the deconvolution in a robust manner. The
prior information is the population-level mean cell-type proportions and
their corresponding standard deviations across samples that are obtained
from scRNA-seq data from <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5228327/" class="external-link">Baron et
al. (2016)</a>.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ave_est</span> <span class="op">=</span> <span class="fu"><a href="../reference/pop.ct.prop.scRNA.html">pop.ct.prop.scRNA</a></span><span class="op">(</span>scRNA<span class="op">=</span><span class="va">baron</span><span class="op">[[</span><span class="st">"sc.eset.qc"</span><span class="op">]</span><span class="op">]</span>,cell_type_unique<span class="op">=</span><span class="va">cell_type_unique</span><span class="op">)</span><span class="op">$</span><span class="va">pop.ct.prop</span></span>
<span><span class="va">ave_sd</span> <span class="op">=</span> <span class="fu"><a href="../reference/pop.ct.prop.scRNA.html">pop.ct.prop.scRNA</a></span><span class="op">(</span>scRNA<span class="op">=</span><span class="va">baron</span><span class="op">[[</span><span class="st">"sc.eset.qc"</span><span class="op">]</span><span class="op">]</span>,cell_type_unique<span class="op">=</span><span class="va">cell_type_unique</span><span class="op">)</span><span class="op">$</span><span class="va">pop.ct.sd</span></span>
<span><span class="va">lambda_option</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span>from<span class="op">=</span><span class="fl">1</span>,to<span class="op">=</span><span class="fl">20</span>,length<span class="op">=</span><span class="fl">4</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span>from<span class="op">=</span><span class="fl">30</span>,to<span class="op">=</span><span class="fl">100</span>,length<span class="op">=</span><span class="fl">4</span><span class="op">)</span>,<span class="fl">200</span>,<span class="fl">500</span>,<span class="fl">1000000</span><span class="op">^</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">InteRD2.output</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/InteRD2.html">InteRD2</a></span><span class="op">(</span>bulk.data<span class="op">=</span><span class="va">pseudo.baron</span><span class="op">$</span><span class="va">pseudo_eset</span>,<span class="va">list_marker</span>,<span class="va">cell_type_unique</span>,comb_sampled<span class="op">=</span><span class="va">InteRD1</span>,<span class="va">ave_est</span>,<span class="va">ave_sd</span>,lambda_option<span class="op">=</span><span class="va">lambda_option</span><span class="op">)</span></span>
<span><span class="va">InteRD2</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/InteRD.predict.prop.html">InteRD.predict.prop</a></span><span class="op">(</span>InteRD.output<span class="op">=</span><span class="va">InteRD2.output</span><span class="op">)</span></span></code></pre></div>
<p>Based on the evaluations, we can tell that InteRD-based estimates
(both InteRD1 and InteRD2) are better than the existing method.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/evaluate.html">evaluate</a></span><span class="op">(</span><span class="va">SCDC_ENSEMBLE_MAD</span>,<span class="va">pseudo.baron</span><span class="op">$</span><span class="va">true_p</span><span class="op">)</span><span class="op">$</span><span class="va">all.eva</span></span>
<span><span class="fu"><a href="../reference/evaluate.html">evaluate</a></span><span class="op">(</span><span class="va">InteRD1</span>,<span class="va">pseudo.baron</span><span class="op">$</span><span class="va">true_p</span><span class="op">)</span><span class="op">$</span><span class="va">all.eva</span></span>
<span><span class="fu"><a href="../reference/evaluate.html">evaluate</a></span><span class="op">(</span><span class="va">InteRD2</span>,<span class="va">pseudo.baron</span><span class="op">$</span><span class="va">true_p</span><span class="op">)</span><span class="op">$</span><span class="va">all.eva</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Chixiang Chen, Yuk Yee Leung, Matei Lonita, Li-San Wang, Mingyao Li.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.5.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
